saleor/payment/gateways/stripe/webhooks.py

```diff
@@ -229,6 +234,10 @@ def handle_authorized_payment_intent(
     if _channel_slug_is_different_from_payment_channel_slug(channel_slug, payment):
         return
 
+    _update_payment_method_metadata(
```

Comment:
You can call `_update_payment_method_metadata` only when you have any `payment.metadata`